packages/box/src/dfuse-receiver.ts

```diff
@@ -0,0 +1,242 @@
+                    ? this.jsonTransactions[this.jsonTransactions.length - 1]
+                    : null;
+            logger.info(
+                `${trx.undo ? "undo block" : "recv block"} ${trx.block.num} ${
+                    trx.trace
+                        ? "trx " + trx.trace.id
+                        : "no matching transactions"
+                }`
+            );
+            if (trx.trace || (prev && prev.trace)) {
+                this.jsonTransactions.push(trx);
+                this.pushTrx(trx);
+                if (
+                    this.jsonTransactions.length - this.numSaved > 10 ||
+                    !trx.trace
+                ) {
+                    logger.info(
+                        `save ${dfuseConfig.jsonTrxFile}: ${this.jsonTransactions.length} transactions and undo entries`
+                    );
+                    fs.writeFileSync(
+                        dfuseConfig.jsonTrxFile + ".tmp",
+                        JSON.stringify(this.jsonTransactions)
+                    );
+                    fs.renameSync(
+                        dfuseConfig.jsonTrxFile + ".tmp",
+                        dfuseConfig.jsonTrxFile
+                    );
+                    this.storage.saveState();
+                    this.numSaved = this.jsonTransactions.length;
+                }
+            }
+            stream.mark({
+                cursor: trx.cursor,
+            });
+        }
+        if (message.type === "error")
+            logger.error(JSON.stringify(message, null, 4));
+    }
+
+    async start() {
+        try {
+            this.jsonTransactions = JSON.parse(
+                fs.readFileSync(dfuseConfig.jsonTrxFile, "utf8")
+            );
+            this.numSaved = this.jsonTransactions.length;
+        } catch (e) {}
+
+        if (this.jsonTransactions.length)
+            this.variables.cursor = this.jsonTransactions[
+                this.jsonTransactions.length - 1
+            ].cursor;
+
+        const begin = performance.now();
+        logger.info("pushing existing blocks...");
+        for (let trx of this.jsonTransactions) this.pushTrx(trx);
+        logger.info(`${performance.now() - begin} ms`);
+        this.storage.saveState();
+
+        if (dfuseConfig.preventConnect) return;
+
+        logger.info("connecting to", dfuseConfig.apiNetwork);
+        if (!this.jsonTransactions.length && dfuseConfig.firstBlock === 1)
+            logger.warn(
+                "Don't have an existing dfuse cursor and DFUSE_FIRST_BLOCK isn't greator than 1; " +
+                    "this may take a while before the first result comes..."
+            );
+
+        const stream = await this.dfuseClient.graphql(
+            query,
+            this.onMessage.bind(this),
+            {
+                operationType: "subscription",
+                variables: this.variables,
+            }
+        );
+
+        logger.info("dfuse is now connected");
+    }
+    catch(e: Error) {
```

Comment:
rename this `catch` method to a non keyword, and add new break line space like you did for the other methods of the class. 

I was actually rereading the code thinking it was a syntax error, like a missing try/catch block...

Reply:
Bizzare that ts didn't complain. The `try` got lost during a refactor.

Reply:
fixed